sale_order_line_packaging_qty/models/sale_order_line.py

```diff
@@ -0,0 +1,108 @@
+    )
+    def _compute_product_packaging_qty(self):
+        for sol in self:
+            if (
+                not sol.product_packaging_id
+                or sol.product_uom_qty == 0
+                or sol.product_packaging_id.qty == 0
+            ):
+                sol.product_packaging_qty = 0
+                continue
+            # Consider uom
+            if sol.product_id.uom_id != sol.product_uom:
+                product_qty = sol.product_uom._compute_quantity(
+                    sol.product_uom_qty, sol.product_id.uom_id
+                )
+            else:
+                product_qty = sol.product_uom_qty
+            # Provide default values in case of missing product.
+            # Odoo does not prevent you to modify product packaging
+            # that are already linked to a sale order.
+            # If by chance the product is removed (yes, you can do it)
+            # this computation will be broken w/out this default.
+            precision = sol.product_packaging_id.product_uom_id.rounding or 0.01
+            qty_mod = float_round(
+                product_qty % sol.product_packaging_id.qty,
+                precision_rounding=precision,
+            )
+            # After the rounding, the value could be equals to
+            # sol.product_packaging_id.qty.
+            # So just re-apply the '%'
+            qty_mod = qty_mod % sol.product_packaging_id.qty
+            if not float_is_zero(
+                qty_mod,
+                precision_digits=precision,
+            ):
+                # If qty does not fit in package reset package qty
+                sol.product_packaging_qty = 0
+            else:
+                # Maybe that product_packaging_qty should be an Integer, no ?
+                qty = product_qty / sol.product_packaging_id.qty
+                sol.product_packaging_qty = qty
+
+    def _prepare_product_packaging_qty_values(self):
+        return {
+            "product_uom_qty": self.product_packaging_id.qty
+            * self.product_packaging_qty,
+            "product_uom": self.product_packaging_id.product_uom_id.id,
+        }
+
+    def _inverse_product_packaging_qty(self):
+        for sol in self:
+            if sol.product_packaging_qty and not sol.product_packaging_id:
+                raise UserError(
+                    _(
+                        "You must define a package before setting a quantity "
+                        "of said package."
+                    )
+                )
+            if sol.product_packaging_id and sol.product_packaging_qty:
+                sol.write(sol._prepare_product_packaging_qty_values())
+
+    @api.onchange("product_packaging_qty")
+    def _onchange_product_packaging_qty(self):
+        if self.product_packaging_id and self.product_packaging_qty:
+            self.update(self._prepare_product_packaging_qty_values())
+
+    @api.onchange("product_packaging_id")
+    def _onchange_product_packaging_id(self):
```

Comment:
This may not make any sense, because product_packaging_id became a computed field in Odoo 16.